lhotse/supervision.py

```diff
@@ -64,6 +66,34 @@ def filter(self, predicate: Callable[[SupervisionSegment], bool]) -> 'Supervisio
         """
         return SupervisionSet.from_segments(seg for seg in self if predicate(seg))
 
+    def find(
+            self,
+            recording_id: str,
+            channel: int,
+            start_after: Seconds = 0,
+            end_before: Optional[Seconds] = None,
+            adjust_offset: bool = False
+    ) -> Iterable[SupervisionSegment]:
+        """
+        Return an iterable of segments that match the provided ``recording_id``.
+
+        :param recording_id: Desired recording ID.
+        :param channel: Return supervisions in the specified channel.
+        :param start_after: When specified, return segments that start after the given value.
+        :param end_before: When specified, return segments that end before the given value.
+        :param adjust_offset: When true, return segments as if the recordings had started at ``start_after``.
+            In other words, :math:`segment.start = segment.start - start_after`.
+        :return: An iterator over supervision segments satisfying all criteria.
+        """
+        return (
+            segment.with_offset(-start_after) if adjust_offset else segment
```

Comment:
can you explain why it's not necessary to modify the duration here?

Reply:
I extended the comments and involved a rationale - hopefully it's clearer now

Reply:
Ah I see.  Could you please add:  Note: in the anticipated use-case, ``start_after`` and ``end_before`` would be the beginning and end of a cut; this option converts the times to be relative to the start of the cut.
IDK why this wasn't clear to me originally, but hopefully this will prevent the same confusion.  Otherwise, LGTM.